src/PSM3.sol

```diff
@@ -5,32 +5,48 @@ import { IERC20 } from "erc20-helpers/interfaces/IERC20.sol";
 
 import { SafeERC20 } from "erc20-helpers/SafeERC20.sol";
 
-import { Math } from "openzeppelin-contracts/contracts/utils/math/Math.sol";
+import { Ownable } from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import { Math }    from "openzeppelin-contracts/contracts/utils/math/Math.sol";
 
 import { IPSM3 }             from "src/interfaces/IPSM3.sol";
 import { IRateProviderLike } from "src/interfaces/IRateProviderLike.sol";
 
-contract PSM3 is IPSM3 {
+contract PSM3 is IPSM3, Ownable {
 
     using SafeERC20 for IERC20;
 
     uint256 internal immutable _asset0Precision;
     uint256 internal immutable _asset1Precision;
     uint256 internal immutable _asset2Precision;
 
-    // NOTE: Assumption is made that asset2 is the yield-bearing counterpart of asset0 and asset1.
-    //       Examples: asset0 = USDC, asset1 = DAI, asset2 = sDAI
+    // NOTE(CRITICAL): Multiple assumptions are made about the initial configuration of this
+    //                 contract that are critical to it functioning correctly.
+    //                 These assumptions are:
+    //                 Assumption 1: `asset2` is the yield-bearing counterpart of asset0 and asset1.
+    //                 Assumption 2: `asset0` is the only asset that can be used with a `pocket`.
+    //                 Example: asset0 = USDC, asset1 = DAI, asset2 = sDAI
+
     IERC20 public override immutable asset0;
```

Comment:
Not related to this PR, but because of the increasing constraints it will be better to rename these to `usdc`, `usds` and `susds` and move away from asset0/1/2. This is to remove any foot guns. I find I'm checking in multiple places if the ordering is correct.

Reply:
Updated